rskj-core/src/test/java/co/rsk/peg/lockingcap/LockingCapSupportImplTest.java

```diff
@@ -0,0 +1,302 @@
+import org.ethereum.core.BlockTxSignatureCache;
+import org.ethereum.core.ReceivedTxSignatureCache;
+import org.ethereum.core.SignatureCache;
+import org.ethereum.core.Transaction;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class LockingCapSupportImplTest {
+    private final LockingCapConstants constants = LockingCapMainNetConstants.getInstance();
+    private LockingCapSupport lockingCapSupport;
+    private LockingCapStorageProvider lockingCapStorageProvider;
+    private ActivationConfig.ForBlock activations = ActivationConfigsForTest.all().forBlock(0);
+    private SignatureCache signatureCache;
+    private StorageAccessor bridgeStorageAccessor;
+
+    @BeforeEach
+    void setUp() {
+        bridgeStorageAccessor = new InMemoryStorage();
+        lockingCapStorageProvider = new LockingCapStorageProviderImpl(bridgeStorageAccessor);
+        signatureCache = new BlockTxSignatureCache(new ReceivedTxSignatureCache());
+        lockingCapSupport = new LockingCapSupportImpl(lockingCapStorageProvider, activations, constants, signatureCache);
+    }
+
+    @Test
+    void getLockingCap_whenNoPreviousValueExists_shouldReturnInitialValue() {
+        // Arrange
+        Optional<Coin> expectedLockingCap = Optional.of(constants.getInitialValue());
+
+        // Act
+        Optional<Coin> actualLockingCap = lockingCapSupport.getLockingCap();
+
+        // Assert
+        assertEquals(expectedLockingCap, actualLockingCap);
+    }
+
+    @Test
+    void getLockingCap_whenAPreviousLockingCapValueExistsInStorage_shouldReturnPreviousValue() {
+        // Arrange
+        Coin previousLockingCap = constants.getInitialValue().add(Coin.SATOSHI);
+        lockingCapStorageProvider.setLockingCap(previousLockingCap);
+        lockingCapSupport.save();
+        // Recreate LockingCapSupport to load the previous value from storage
+        lockingCapStorageProvider = new LockingCapStorageProviderImpl(bridgeStorageAccessor);
+        lockingCapSupport = new LockingCapSupportImpl(lockingCapStorageProvider, activations, constants, signatureCache);
+
+        // Act
+        Optional<Coin> actualLockingCap = lockingCapSupport.getLockingCap();
+
+        // Assert
+        assertEquals(Optional.of(previousLockingCap), actualLockingCap);
+    }
+
+    @Test
+    void getLockingCap_prePapyrus200_whenLockingCapIsNotSet_shouldReturnEmpty() {
+        // Arrange
+        activations = ActivationConfigsForTest.wasabi100().forBlock(0);
+        lockingCapSupport = new LockingCapSupportImpl(lockingCapStorageProvider, activations, constants, signatureCache);
+
+        // Act
+        Optional<Coin> actualLockingCap = lockingCapSupport.getLockingCap();
+
+        // Assert
+        assertEquals(Optional.empty(), actualLockingCap);
+    }
+
+    @Test
+    void increaseLockingCap_whenNoPreviousValueExistsAndTakeDefaultInitialValueAndNewLockingCapIsGreaterThanInitialValue_shouldReturnTrue() {
```

Comment:
i would just say `_whenNewValueIsGreaterThanCurrentLockingCap`, no need to explain the initial value logic in the test name.

Reply:
Done!